wave_kinematics/src/rotation.cpp

```diff
@@ -0,0 +1,99 @@
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/utils.hpp"
+
+double wave::Rotation::comparision_threshold = 1e-8;
+
+wave::Rotation::Rotation() {}
+
+void wave::Rotation::setToIdentity() {
+  this->rotation_object.setIdentity();
+}
+
+void wave::Rotation::compose(const Rotation &R) {
+  this->rotation_object = this->rotation_object * R.rotation_object;
```

Comment:
I'll implement an overloaded * operator, but those are mainly for convenience.  The point of having a dedicated compose expression is that there will also be a matching composeAndJacobian, which is much stricter on defining a **right** sided composition, and therefore Jacobians of the expression with respect to the left rotation, and Jacobians wrt the right rotation.  The big picture here, is that the kinematics library supplies Jacobians of these elementary operations, such that when we go to implement our residual functions, the Jacobian of the residual is easily calculated through chain rule of the elementary functions.  

I believe we should get into the habit of viewing these residual expressions as a series of composed functions, which I think is easier to do when explicitly in function form, rather than using operators.